demo/blocks_update/run.py

```diff
@@ -1,17 +1,44 @@
-from gradio import Dropdown, Blocks, Slider
+import gradio as gr
 
+with gr.Blocks() as block:
+    gr.Markdown(
+        """
+    # Animal Generator
+    Once you select a species, the detail panel should be visible.
+    """
+    )
 
-def update_dropdown_choices(val):
-    if val < 50:
-        opts = ["fail", "repeat"]
-    else:
-        opts = ["succeed", "repeat"]
-    return Dropdown.update(choices=opts)
+    species = gr.Radio(label="Animal Class", choices=["Mammal", "Fish", "Bird"])
+    animal = gr.Dropdown(label="Animal", choices=[])
 
+    with gr.Column(visible=False) as details_col:
+        weight = gr.Slider(0, minimum=0, maximum=20)
+        details = gr.Textbox(label="Extra Details")
+        generate_btn = gr.Button("Generate")
+        output = gr.Textbox(label="Output")
 
-with Blocks() as block:
-    s = Slider(default_value=0, minimum=0, maximum=100)
-    d = Dropdown(choices=["fake", "choices", "exist"])
+    species_map = {
+        "Mammal": ["Elephant", "Giraffe", "Hamster"],
+        "Fish": ["Shark", "Salmon", "Tuna"],
+        "Bird": ["Chicken", "Eagle", "Hawk"],
+    }
 
-    s.change(fn=update_dropdown_choices, inputs=[s], outputs=[d])
-    block.launch()
+    def filter_species(species):
+        return gr.Dropdown.update(
+            choices=species_map[species], value=species_map[species][1]
+        ), gr.update(visible=True)
+
+    species.change(filter_species, species, [animal, details_col])
```

Comment:
I have an idea which will unlock more opportunities with the update functionality, but might make things a little bit more complex from the development POV.

what if we send components in to the function and update component's field over there?
```

    def filter_species(species, dropdown, column):
        return (dropdown.update(
            choices=species_map[species], value=species_map[species][1]
        )
        , column.update(visible=True))
      
       species.update_on_change(filter_species, [species, animal, details_col], [animal, details_col])
```

We would need to distinguish between sending values inside the component and sending the component itself. But this would enable better intution, and fetching a Component's fields and updating component's fields according to the previous fields.

Reply:
This is a pretty drastic change, and I'm not sure if it makes the API cleaner. I think there are pros/cons to it but maybe discussion for after 3.0?